src/audio/eq_iir/eq_iir.c

```diff
@@ -216,6 +233,9 @@ static void eq_iir_s32_16_default(const struct comp_dev *dev, const struct audio
 	const int samples = frames * nch;
 	int processed = 0;
 
+	bsource->consumed += samples << 2;
+	bsink->size += samples << 1;
```

Comment:
This is more special so, separate macros of functions for source and sink when different format.